tests/e2e/utils/device.go

```diff
@@ -0,0 +1,1224 @@
+			Modbus: &v1alpha1.VisitorConfigModbus{
+				Register:       "CoilRegister",
+				Offset:         2,
+				Limit:          1,
+				Scale:          1,
+				IsSwap:         true,
+				IsRegisterSwap: true,
+			},
+		},
+	}
+	devicePropertyVisitor2 := v1alpha1.DevicePropertyVisitor{
+		PropertyName: "temperature-enable",
+		VisitorConfig: v1alpha1.VisitorConfig{
+			Modbus: &v1alpha1.VisitorConfigModbus{
+				Register:       "DiscreteInputRegister",
+				Offset:         3,
+				Limit:          1,
+				Scale:          1.0,
+				IsSwap:         true,
+				IsRegisterSwap: true,
+			},
+		},
+	}
+	propertyVisitors := []v1alpha1.DevicePropertyVisitor{devicePropertyVisitor1, devicePropertyVisitor2}
+	newDeviceModel := v1alpha1.DeviceModel{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "DeviceModel",
+			APIVersion: "devices.kubeedge.io/v1alpha1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      "sensor-tag-model",
+			Namespace: Namespace,
+		},
+		Spec: v1alpha1.DeviceModelSpec{
+			Properties:       properties,
+			PropertyVisitors: propertyVisitors,
+		},
+	}
+	return newDeviceModel
+}
+
+func NewBluetoothDeviceModel() v1alpha1.DeviceModel {
+	deviceProperty1 := v1alpha1.DeviceProperty{
+		Name:        "temperature",
+		Description: "temperature in degree celsius",
+		Type: v1alpha1.PropertyType{Int: &v1alpha1.PropertyTypeInt64{
+			AccessMode: "ReadOnly",
+			Maximum:    100,
+			Unit:       "degree celsius",
+		}},
+	}
+	deviceProperty2 := v1alpha1.DeviceProperty{
+		Name:        "temperature-enable",
+		Description: "enable data collection of temperature sensor",
+		Type: v1alpha1.PropertyType{String: &v1alpha1.PropertyTypeString{
+			AccessMode:   "ReadWrite",
+			DefaultValue: "ON",
+		}},
+	}
+	deviceProperty3 := v1alpha1.DeviceProperty{
+		Name:        "io-config-initialize",
+		Description: "initialize io-config with value 0",
+		Type: v1alpha1.PropertyType{Int: &v1alpha1.PropertyTypeInt64{
+			AccessMode:   "ReadWrite",
+			DefaultValue: 0,
+		}},
+	}
+	deviceProperty4 := v1alpha1.DeviceProperty{
+		Name:        "io-data-initialize",
+		Description: "initialize io-data with value 0",
+		Type: v1alpha1.PropertyType{Int: &v1alpha1.PropertyTypeInt64{
+			AccessMode:   "ReadWrite",
+			DefaultValue: 0,
+		}},
+	}
+	deviceProperty5 := v1alpha1.DeviceProperty{
+		Name:        "io-config",
+		Description: "register activation of io-config",
+		Type: v1alpha1.PropertyType{Int: &v1alpha1.PropertyTypeInt64{
+			AccessMode:   "ReadWrite",
+			DefaultValue: 1,
+		}},
+	}
+	deviceProperty6 := v1alpha1.DeviceProperty{
+		Name:        "io-data",
+		Description: "data field to control io-control",
+		Type: v1alpha1.PropertyType{Int: &v1alpha1.PropertyTypeInt64{
+			AccessMode:   "ReadWrite",
+			DefaultValue: 0,
+		}},
+	}
+	properties := []v1alpha1.DeviceProperty{deviceProperty1, deviceProperty2, deviceProperty3, deviceProperty4, deviceProperty5, deviceProperty6}
+	devicePropertyVisitor1 := v1alpha1.DevicePropertyVisitor{
```

Comment:
We should reduce writing code for constructing objects like properties or visitors.

Reply:
this applies to all the code throughout . let's improve in the next iteration.

Reply:
sure